chain-ser/src/deser.rs

```diff
@@ -84,3 +94,75 @@ impl<const N: usize> Deserialize for [u8; N] {
         Ok(buf)
     }
 }
+
+impl SerializedSize for u8 {
+    fn serialized_size(&self) -> usize {
+        std::mem::size_of::<u8>()
+    }
+}
+
+impl SerializedSize for u16 {
+    fn serialized_size(&self) -> usize {
+        std::mem::size_of::<u16>()
+    }
+}
+
+impl SerializedSize for u32 {
+    fn serialized_size(&self) -> usize {
+        std::mem::size_of::<u32>()
+    }
+}
+
+impl SerializedSize for u64 {
+    fn serialized_size(&self) -> usize {
+        std::mem::size_of::<u64>()
+    }
+}
+
+impl SerializedSize for u128 {
```

Comment:
How can you define the serialized size without knowing the `Serialize` implementation or the format used? Big-endian or little-endian are not the only possible formats, what if the implementor decide to use something like https://docs.rs/bincode/latest/bincode/config/struct.VarintEncoding.html?